packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelFactoryProvider.cs

```diff
@@ -129,8 +130,8 @@ private static IReadOnlyList<ValueExpression> GetCtorArgs(
                 }
             }
 
-            var modelContainsAdditionalRawData = modelCtorFullSignature.Parameters.Any(p => p.Name.Equals(AdditionalRawDataParameterName));
-            if (modelContainsAdditionalRawData)
+            var modelContainsAdditionalRawData = modelCtorFullSignature.Parameters.Any(p => p.Name.Equals(AdditionalBinaryDataParameterName));
+            if (modelContainsAdditionalRawData && !supportsBinaryDataAdditionalProps)
```

Comment:
Why do we need the check `!supportsBinaryDataAdditionalProps`?

Reply:
This is to default that parameter value to `null` in the case that a model doesn't explicitly support additional properties of BinaryData / unknown type.